src/NServiceBus.Core/Encryption/ConfigureRijndaelEncryptionService.cs

```diff
@@ -169,12 +169,14 @@ IList<byte[]> expiredKeys
         public static void RegisterEncryptionService(this EndpointConfiguration config, Func<IEncryptionService> func)
         {
             Guard.AgainstNull(nameof(config), config);
-            config.Settings.Set("EncryptionServiceConstructor", func);
+
+            config.EnableFeature<Encryptor>();
```

Comment:
instead of manually enabling the feature on the settings which trigger activation, couldn't we check for the setting value on the feature setup as a prerequisite and let the feature disable itself?